pkg/controller/performanceprofile/performanceprofile_controller.go

```diff
@@ -272,14 +264,41 @@ func (r *ReconcilePerformanceProfile) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, nil
 	}
 
+	// TODO: we need to check if all under performance profiles values != nil
+	// first we need to decide if each of values required and we should move the check into validation webhook
+	// for now let's assume that all parameters needed for assets scrips are required
+	if err := r.validatePerformanceProfileParameters(instance); err != nil {
+		klog.Errorf("failed to reconcile: %v", err)
+		conditions := r.getDegradedConditions(conditionReasonValidationFailed, err.Error())
+		if err := r.updateStatus(instance, conditions); err != nil {
+			klog.Errorf("failed to update performance profile %q status: %v", instance.Name, err)
+			return reconcile.Result{}, err
+		}
+		// we do not want to reconcile again in case of error, because a user will need to update the PerformanceProfile anyway
+		return reconcile.Result{}, nil
+	}
+
 	// apply components
 	result, err := r.applyComponents(instance)
 	if err != nil {
-		klog.Errorf("failed to deploy components: %v", err)
+		klog.Errorf("failed to deploy performance profile %q components: %v", instance.Name, err)
```

Comment:
you could collapse all this if/else logic and just pass the applyComponents error result to updateStatus. applyComponents would need to always return a results struct instead of nil though for this to work.

so

```
result, syncErr := applyComponents(instance)
if syncErr != nil {
    klog.Errorf("failed to deploy performance profile %q components: %v", instance.Name, syncErr)
}

if err := r.updateStatus(instance, syncErr); err != nil {
    klog.Errorf("failed to update performance profile %q status: %v", instance.Name, err)
    return reconcile.Result{}, err
}

return *result, syncErr
```

updateStatus would then set the degraded condition if syncErr != nil